applications/tari_base_node/src/builder.rs

```diff
@@ -53,20 +79,200 @@ impl NodeType {
+                warn!(
+                    target: LOG_TARGET,
+                    "{} is not a valid peer seed. The public key is incorrect. {}",
+                    s,
+                    e.to_string()
+                );
+                continue;
+            },
+            Ok(p) => p,
+        };
+        let addr = match parts[1].parse::<NetAddress>() {
+            Err(e) => {
+                warn!(
+                    target: LOG_TARGET,
+                    "{} is not a valid peer seed. The address is incorrect. {}",
+                    s,
+                    e.to_string()
+                );
+                continue;
+            },
+            Ok(a) => a,
+        };
+        let node_id = match NodeId::from_key(&pub_key) {
+            Err(e) => {
+                warn!(
+                    target: LOG_TARGET,
+                    "{} is not a valid peer seed. A node id couldn't be derived from the public key. {}",
+                    s,
+                    e.to_string()
+                );
+                continue;
+            },
+            Ok(id) => id,
+        };
+        let peer = Peer::new(
+            pub_key,
+            node_id,
+            NetAddressesWithStats::from(addr),
+            PeerFlags::default(),
+            PeerFeatures::empty(),
+        );
+        result.push(peer);
+    }
+    result
+}
+
+fn setup_comms_services<T: BlockchainBackend + 'static>(
+    rt: &Runtime,
+    id: Arc<NodeIdentity>,
+    peers: Vec<Peer>,
+    peer_db_path: &str,
+    db: BlockchainDatabase<T>,
+) -> OutboundNodeCommsInterface
+{
+    let node_config = BaseNodeServiceConfig::default(); // TODO - make this configurable
+    let (publisher, subscription_factory) = pubsub_connector(rt.executor(), 100);
+    let subscription_factory = Arc::new(subscription_factory);
+    let comms_config = CommsConfig {
+        node_identity: id.clone(),
+        host: "127.0.0.1".parse().unwrap(),
+        socks_proxy_address: None,
+        control_service: ControlServiceConfig {
+            listener_address: id.control_service_address(),
+            socks_proxy_address: None,
+            requested_connection_timeout: Duration::from_millis(2000),
+        },
+        datastore_path: peer_db_path.to_string(),
+        peer_database_name: "peers".to_string(),
+        inbound_buffer_size: 100,
+        outbound_buffer_size: 100,
+        dht: Default::default(),
+    };
+
+    let (comms, dht) = initialize_comms(rt.executor(), comms_config, publisher).unwrap();
+
+    for p in peers {
+        debug!(target: LOG_TARGET, "Adding seed peer [{}]", p.node_id);
+        comms.peer_manager().add_peer(p).unwrap();
+    }
+
+    let fut = StackBuilder::new(rt.executor(), comms.shutdown_signal())
+        .add_initializer(CommsOutboundServiceInitializer::new(dht.outbound_requester()))
+        .add_initializer(BaseNodeServiceInitializer::new(
+            subscription_factory,
+            id.clone(),
+            Arc::new(db),
```

Comment:
The BaseNodeServiceInitializer was changed in a recently merged PR and it doesn't need an Arc for the db.

Reply:
:+1: